.rubocop.yml

```diff
@@ -1,3 +1,12 @@
+inherit_from:
+  - ./config/.lint_rubocop.yml
+  - ./config/.metrics_rubocop.yml
+  - ./config/.rails_rubocop.yml
+  - ./config/.style_rubocop.yml
+  - ./config/.layout_rubocop.yml
+  - ./config/.naming_rubocop.yml
+  - ./config/.security_rubocop.yml
```

Comment:
Have divided rules into their own files, to keep things more cleaner. But this way have a drawback with codeclimate. Codeclimate when it prepares files, it will only import this path, but not the content in those files.

So when we are going to import this into app repo, it has to be done like:

```yml
prepare:
  fetch:
    - url: "https://raw.githubusercontent.com/shiftcommerce/shift-rubocop/feature/update_rubocop_rules/.rubocop.yml"
      path: ".rubocop.yml"
    - url: "https://raw.githubusercontent.com/shiftcommerce/shift-rubocop/feature/update_rubocop_rules/config/.lint_rubocop.yml"
      path: "config/.lint_rubocop.yml"
    - url: "https://raw.githubusercontent.com/shiftcommerce/shift-rubocop/feature/update_rubocop_rules/config/.metrics_rubocop.yml"
      path: "config/.metrics_rubocop.yml"
    - url: "https://raw.githubusercontent.com/shiftcommerce/shift-rubocop/feature/update_rubocop_rules/config/.layout_rubocop.yml"
      path: "config/.layout_rubocop.yml"
    - url: "https://raw.githubusercontent.com/shiftcommerce/shift-rubocop/feature/update_rubocop_rules/config/.rails_rubocop.yml"
      path: "config/.rails_rubocop.yml"
    - url: "https://raw.githubusercontent.com/shiftcommerce/shift-rubocop/feature/update_rubocop_rules/config/.style_rubocop.yml"
      path: "config/.style_rubocop.yml"
    - url: "https://raw.githubusercontent.com/shiftcommerce/shift-rubocop/feature/update_rubocop_rules/config/.naming_rubocop.yml"
      path: "config/.naming_rubocop.yml"
    - url: "https://raw.githubusercontent.com/shiftcommerce/shift-rubocop/feature/update_rubocop_rules/config/.security_rubocop.yml"
      path: "config/.security_rubocop.yml"
```

This explicit mentioning of files is also again due to codeclimate limitations. This is the message we have from codeclimate team:

` What's happening is that we're unable to use the inherit_from directive in your .rubocop.yml. We do hope to eventually support importing base config files directly from tools like Rubocop, but we haven't figured out a good way to do it yet that doesn't break aspects of the limits we place on engines for security reasons. Unfortunately, there's no way to recursively import the config files you mentioned; the explicit imports you're currently using are your best option for now. Sorry about the limitation with that. `